os/src/main/java/io/smallrye/common/os/Process.java

```diff
@@ -29,6 +29,10 @@
  * @author <a href="mailto:david.lloyd@redhat.com">David M. Lloyd</a>
  */
 public final class Process {
+    private static final ProcessHandle current = doPrivileged((PrivilegedAction<ProcessHandle>) ProcessHandle::current);
```

Comment:
Do we actually need to keep this one around.
From what I can see, it's only used to get the PID?

I wonder if we should have a private `record` with the state (pid, process name, command) and only cache this?

`ProcessHandle` is out of our reach and I'm very nervous making it static.

Reply:
When you call `ProcessHandle.current()`, it's returning a static final field already. The object itself is a holder for a PID value (`long`)/start time (another `long`) combination, so generally it's safe to retain (especially _our own_ PID). 

What we're caching here is basically the side-effects of calling through `doPrivileged`, which is still required for WildFly for a little while longer. Once that's no longer needed by any downstream projects, we can replace it with calls to `ProcessHandle.current()`.

Reply:
OK, all good then, thanks for the details.

I have probably spent too much time looking at heap dumps and CL leaks, lately :).